libnetwork/network/interface.go

```diff
@@ -137,24 +137,30 @@ func defaultNetworkBackend(store storage.Store, conf *config.Config) (backend ty
 	if err != nil {
 		return "", err
 	}
-	if len(cons) == 0 {
-		imgs, err := store.Images()
-		if err != nil {
-			return "", err
-		}
-		if len(imgs) == 0 {
+	if len(cons) != 0 {
+		return types.CNI, nil
+	}
+
+	imgs, err := store.Images()
+	if err != nil {
+		return "", err
+	}
+	// ignore ReadOnly images
+	for _, i := range imgs {
+		if !i.ReadOnly {
 			cniInterface, err := getCniInterface(conf)
 			if err == nil {
 				nets, err := cniInterface.NetworkList()
 				// there is always a default network so check <= 1
-				if err == nil && len(nets) <= 1 {
+				if err == nil || len(nets) <= 1 {
```

Comment:
this would always be true since there is very likely no error

Reply:
Yes this was a bug, in the rewrite.